R/optionalInput.R

```diff
@@ -212,7 +214,7 @@ optionalSelectInput <- function(inputId, # nolint
     shinyWidgets::pickerInput(
       inputId = inputId,
       label = label,
-      choices = raw_choices,
+      choices = if (inherits(raw_choices, "function")) raw_choices() else raw_choices,
```

Comment:
isn't `raw_choices` a function of `data`? I think so, it means this change introduces a bug, as evaluation `raw_choices()` will throw an error

Reply:
this whole PR was wrong, I will write what was the fix on the original issue https://github.com/insightsengineering/teal.goshawk/issues/301